lib/smart_app_launch/token_introspection_response_group.rb

```diff
@@ -0,0 +1,127 @@
+        the tester will need to run an out-of-band ACCESS TOKEN request and manually input the access token response
+        parameters.   
+        
+        2. The token introspection response bodies. If the Token Introspection Request test group was run, these will
+        auto-fill; otherwise, the tester will need to an run out-of-band INTROSPECTION requests for:
+
+          a. The ACTIVE access token received from the out-of-band access token request, AND
+
+          b. An INACTIVE OR INVALID token 
+
+        The client making both introspection requests must be authorized to access the introspection endpoint if
+        the endpoint is protected.
+      )
+      
+    test do 
+      title 'Token introspection response for an active token contains required fields'
+
+      description %(
+        This test will check whether the metadata in the token introspection response is correct for an active token and
+         that the response data matches the data in the original access token and/or access token response from the 
+         authorization server, including the following:
+      
+        Required:
+        *  `active` claim is set to true 
+        * `scope`, `client_id`, and `exp` claim(s) match between introspection response and access token
+
+        Conditionally Required:
+        * IF launch context parameter(s) included in access token, introspection response includes claim(s) for 
+        launch context parameter(s) 
+          * Parameters checked for are `patient` and `encounter`
+        * IF identity token was included as part of access token response, `iss` and `sub` claims are present in 
+        introspection response
+
+        Optional but Recommended:
+        * IF identity token was included as part of access token response, `fhirUser` claim SHOULD be present in 
+        introspection response
+      )
+
+      # Access token value is repeatedly overwritten during refresh token tests and value 
+      # when provided for introspection is from the last refresh token exchange test from standalone launch
+      # But other outputs (id token, received scopes, etc) do not update and are still the same as
+      # they were in the original access token request 
+      # Is there any contexts in which this could cause problems, i.e., the response body values from 
+      # using a refresh token grant are different than the response body values from the authorization code grant?
+
+      input :standalone_client_id,
+            title: 'Access Token Respone: client_id'
+
+      # TODO delete when verification done
+      input :standalone_access_token,
+            title: 'Access Token Introspected',
+            locked: true,
+            description: 'Included to ensure inputs are populating correctly, will delete later'
+      
+      input :standalone_expires_in,
+            title: 'Access Token Response: expires_in',
```

Comment:
I maybe would phrase this as 'Expected expires_in'?  Can we even verify this?  https://datatracker.ietf.org/doc/html/rfc7662#section-2.2

Reply:
Both `client_id` and `exp` requirements from the [Token Introspection IG](https://hl7.org/fhir/smart-app-launch/token-introspection.html#required-fields-in-the-introspection-response) are erroneously written in my opinion.  Perhaps I am taking them too literally, but they suggest that the values for `client_id` and `exp` fields in the introspection response should match the values for `client_id` and `exp` fields from the access token response.  But, as you said, neither of those are actually present in the access token response (unless the access token is a JWT, which defeats the entire purpose of token introspection...).  Furthermore, the correct value for `exp` (as an integer timestamp) cannot be verified with the information we have access to.  My suggestion for this is to simply check that it is included.